### PR TITLE
introduce Translate API

### LINKED ISSUE #977: Add support for the Translate API
https://cloud.google.com/translate/docs

REST API: https://cloud.google.com/translate/v2/using_rest

### PR DESCRIPTION
**Pull request by stephenplusplus**
Fixes #977 
#### To Dos
- [x] Enable Translate API for Travis account
- [x] Set env var in Travis for API key
- [x] Docs
- [x] Tests
  - [x] Unit
  - [x] System
#### Preview

``` js
var gcloud = require('gcloud');

var translate = gcloud.translate({
  key: 'API Key'
});

// Translate a string of text.
translate.translate('Hello', 'es', function(err, translation) {
  if (!err) {
    // translation = 'Hola'
  }
});

// Detect a language from a string of text.
translate.detect('Hello', function(err, results) {
  if (!err) {
    // results = {
    //   language: 'en',
    //   confidence: 1,
    //   input: 'Hello'
    // }
  }
});

// Get a list of supported languages.
translate.getLanguages(function(err, languages) {
  if (!err) {
    // languages = [
    //   'af',
    //   'ar',
    //   'az',
    //   ...
    // ]
  }
});
```

### CODE REVIEW

**Comment by stephenplusplus** [comment]
@callmehiphop PTAQL. This one's pretty small. I just want to be sure I'm not doing anything too crazy before locking it up with the unit tests.

File: lib/translate/index.js [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+ * translate.translate(input, 'es', function(err, translations) {
+ *   if (!err) {
+ *     // translations = [
+ *     //   'Hola',
+ *     //   'Como estas hoy?'
+ *     // ]
+ *   }
+ * });
+ */
+Translate.prototype.translate = function(input, language, callback) {
```
Comment (callmehiphop) [inline lib/translate/index.js]: `translate.translate(...)` reads kind of funny to me, thoughts on using a different verb..? I'm not too worried about it though, so feel free to ignore :)

**Comment by callmehiphop** [comment]
@stephenplusplus LGTM :) unit test away!

**Comment by stephenplusplus** [comment]
Thanks for the review!

File: lib/translate/index.js (hunk 1 above)
Comment (stephenplusplus) [inline lib/translate/index.js]: I didn't say anything about it to see if it bugged you too :) I'm struggling to think of something better. Super open to it though!

Comment (callmehiphop) [inline lib/translate/index.js]: Yeah, might as well just leave it as translate, I thought maybe convert would be an ok name, but it's pretty much just a synonym for translate.

File: README.md [hunk 2]
```diff
@@ -337,6 +338,54 @@ zone.createVM(name, { os: 'ubuntu' }, function(err, vm, operation) {
...
+
+**An API key is required for Translate.** See [Identifying your application to Google][api-key-howto].
+
+#### Preview
+
+```js
+var gcloud = require('gcloud');
+
+var translate = gcloud.translate({
+  key: 'API Key'
```
Comment (jgeewax) [inline README.md]: Is there no way to use Translate with the standard keyfile.json ?

File: README.md [hunk 3]
```diff
@@ -337,6 +338,54 @@ zone.createVM(name, { os: 'ubuntu' }, function(err, vm, operation) {
...
+});
+
+// Translate a string of text.
+translate.translate('Hello', 'es', function(err, translation) {
```
Comment (jgeewax) [inline README.md]: Is there a way to set the origin language as well as the target language? Looks like this auto-detects the origin (as 'en') and then sends it back in the 'es' as the target.

File: README.md (hunk 2 above)
Comment (stephenplusplus) [inline README.md]: Not that I could figure out. Can you check if and how that would work?

File: README.md (hunk 3 above)
Comment (stephenplusplus) [inline README.md]: Yep, this uses auto-detect. I'll see about opening it up for more options.

Comment (jgeewax) [inline README.md]: Cool maybe it's:

```js
translate.translate('Hello', 'es', callback);
// and then
translate.translate('Hello', {from: 'en', to: 'es'}, callback);
```
?

Comment (stephenplusplus) [inline README.md]: Perfect! Writing it up now.

Comment (callmehiphop) [inline README.md]: :+1:

Comment (stephenplusplus) [inline README.md]: https://github.com/GoogleCloudPlatform/gcloud-node/commit/395b59091059008580172f33f449a2086525de61

**Comment by jgeewax** [comment]
/cc @ilsung

Does Translate work with service account credentials (ie, keyfile.json) ?

**Comment by stephenplusplus** [comment]
Tests added: https://github.com/stephenplusplus/gcloud-node/commit/1ebcde091bb19cb24b9dca9ebe4fc6fbc8efd57b

**Comment by stephenplusplus** [comment]
Marking as blocked until we hear from @ilsung about authenticating with a service account keyfile. (Thanks in advance for your help!)

**Comment by ilsung** [comment]
Sorry for the delay -- No, the Translate API does not support service account creds since it doesn't use OAuth.  A key is required.

**Comment by stephenplusplus** [comment]
Thanks for clearing that up.

@callmehiphop if this LGTY, merge away.

**Comment by callmehiphop** [comment]
:cake:

**Pull request merged**